### PR TITLE
[SDPAP-7477]add new field content collection 

### PR DESCRIPTION
**Pull request by yeniatencio**
**JIRA issue:**
- https://digital-vic.atlassian.net/browse/SDPAP-7477
- https://digital-vic.atlassian.net/browse/SDPAP-7476
BE testing link: https://nginx-php.pr-272.content-reference-sdp-vic-gov-au.sdp4.sdp.vic.gov.au/
FE testing link: https://app.pr-41.reference-sdp-vic-gov-au.sdp4.sdp.vic.gov.au/
### Changed

1. Limit radio buttons to 3 options once added them as mandatory.
2. Updated the massageFormValues function to store the new value as per json schema.
3. New url field (enabled CTA that was disabled during phase 1)
4. Updated default text.
5. Updated field required side-effect.

### Screenshots
Screenshot 2023-05-29 at 6 58 29 pm
Screenshot 2023-05-29 at 6 58 17 pm

JSON for new url field
image

### CODE REVIEW

**Review by MdNadimHossain — APPROVED** [review]
Looks good to me, but add a suggestion for the testers to test it for existing components and see how does it behave.

**Comment by yeniatencio** [comment]
> Looks good to me, but add a suggestion for the testers to test it for existing components and see how does it behave.

Will do. Thanks @MdNadimHossain

File: modules/tide_content_collection/src/Plugin/Field/FieldWidget/ContentCollectionConfigurationWidget.php [hunk 1]
```diff
@@ -505,8 +505,13 @@ public function formElement(FieldItemListInterface $items, $delta, array $elemen
       $element['callToAction']['text'] = [
         '#type'  => 'textfield',
         '#title'  => $this->t('Text'),
-        '#default_value' => $json_object['callToAction']['text'] ?? '',
+        '#default_value' => $json_object['callToAction']['text'] ?? 'View all',
```
Comment (edyuenyw) [inline modules/tide_content_collection/src/Plugin/Field/FieldWidget/ContentCollectionConfigurationWidget.php:508]: For URL new field changes.

File: modules/tide_content_collection/src/Plugin/Field/FieldWidget/ContentCollectionConfigurationWidget.php [hunk 2]
```diff
@@ -505,8 +505,13 @@ public function formElement(FieldItemListInterface $items, $delta, array $elemen
...
         '#description' => $this->t('Display text of the link.'),
+        '#states' => [
+          'required' => [
+            ':input[name="' . $this->getFormStatesElementName('callToAction|url', $items, $delta, $element) . '"]' => ['filled' => TRUE],
+          ],
+        ],
```
Comment (edyuenyw) [inline modules/tide_content_collection/src/Plugin/Field/FieldWidget/ContentCollectionConfigurationWidget.php:514]: For URL new field changes.

File: modules/tide_content_collection/tide_content_collection.install [hunk 3]
```diff
@@ -97,3 +97,15 @@ function tide_content_collection_update_8002() {
...
+ * Enable CTA url field [SDPAP-7475].
+ */
+function tide_content_collection_update_8003() {
+  $config = \Drupal::configFactory()
+    ->getEditable('core.entity_form_display.paragraph.content_collection_enhanced.default');
+  if ($config) {
+    $config->set('content.field_content_collection_config.settings.content.enable_call_to_action', TRUE);
+    $config->save();
+  }
+}
```
Comment (edyuenyw) [inline modules/tide_content_collection/tide_content_collection.install:111]: For URL new field changes.

File: modules/tide_content_collection/config/install/core.entity_form_display.paragraph.content_collection_enhanced.default.yml [hunk 4]
```diff
@@ -34,7 +34,7 @@ content:
     weight: 2
     settings:
       content:
-        enable_call_to_action: false
+        enable_call_to_action: true
```
Comment (edyuenyw) [inline modules/tide_content_collection/config/install/core.entity_form_display.paragraph.content_collection_enhanced.default.yml:37]: For URL new field changes.

**Comment by edyuenyw** [comment]
Hi @MdNadimHossain can you re-review further changes to another ticket relating to CC new URL field. "I've add comments on those updates ("For URL new field changes."). Thanks

File: dev-tools.sh [hunk 5]
```diff
@@ -33,6 +33,6 @@
 # stable version.
 #
 # Uncomment and set the Dev-Tools's commit value and commit this change.
-# export GH_COMMIT=COMMIT_SHA
+export GH_COMMIT=70b73d6c169cbc457351745634105402ab326a27
```
Comment (MdNadimHossain) [inline dev-tools.sh:36]: Revert this change before merging please

**Review by MdNadimHossain — APPROVED** [review]
